src/allocator.cpp

```diff
@@ -33,6 +33,7 @@ class PoolAllocatorPrivate
     Mutex budgets_lock;
     Mutex payouts_lock;
     unsigned int size_compare_ratio; // 0~256
+    static const size_t size_threshold = 10;
```

Comment:
`size_threshold` does not seem expressive enough
`budget_count_threshold` or `budget_drop_threshold` might be better?

How about adding a setter api that allows the user to control whether a more aggressive or conservative budget recycling strategy is required?

Reply:
Renamed & Added a setter.